app/models/agents/website_agent.rb

```diff
@@ -259,13 +260,19 @@ def check_urls(in_url)
       end
     end
 
+    def get_valid_url(url)
+      return (url =~ URI::ABS_URI) ? url : URI.escape(url)
```

Comment:
@knu  [the docs](http://ruby-doc.org/stdlib-2.2.2/libdoc/uri/rdoc/URI/Escape.html) don't mention it being obsolete but [stackoverflow questions](http://stackoverflow.com/a/13059657/4397001) do. Am I opening a can of worms by implementing it myself or should I go for something like [Addressable gem](http://www.rubydoc.info/gems/addressable/2.2.4/Addressable/URI#encode-class_method)?